provider/k8s.go

```diff
@@ -457,18 +625,182 @@ func (k8s *K8sOfflineStore) RegisterPrimaryFromSourceTable(id ResourceID, source
+	}
+	k8s.logger.Debugw("Running SQL transformation", config)
+	runnerArgs := k8s.pandasRunnerArgs(transformationDestination, updatedQuery, sources, Transform)
+	if err := k8s.executor.ExecuteScript(runnerArgs); err != nil {
+		k8s.logger.Errorw("job for transformation failed to run", config.TargetTableID, err)
+		return fmt.Errorf("job for transformation %v failed to run: %v", config.TargetTableID, err)
+	}
+	k8s.logger.Debugw("Succesfully ran SQL transformation", config)
+	return nil
+}
+
+func (k8s *K8sOfflineStore) dfTransformation(config TransformationConfig, isUpdate bool) error {
+	transformationDestination := blobResourcePath(config.TargetTableID)
+	// latestKey//bm22
+	exists, err := k8s.store.Exists(transformationDestination)
+	if err != nil {
+		k8s.logger.Errorw("Error checking if resource exists", err)
+		return err
+	}
+
+	if !isUpdate && exists {
+		k8s.logger.Errorw("Transformation already exists", config.TargetTableID, transformationDestination)
+		return fmt.Errorf("transformation %v already exists at %s", config.TargetTableID, transformationDestination)
+	} else if isUpdate && !exists {
+		k8s.logger.Errorw("Transformation doesn't exists at destination and you are trying to update", config.TargetTableID, transformationDestination)
+		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination)
+	}
+
+	transformationFilePath := k8s.store.ResourcePath(config.TargetTableID)
+	fileName := "transformation.pkl"
+	transformationFileLocation := fmt.Sprintf("%s/%s", transformationFilePath, fileName)
+	err = k8s.store.Write(transformationFileLocation, config.Code)
+	if err != nil {
+		return fmt.Errorf("could not upload file: %s", err)
+	}
+
+	k8sArgs, err := k8s.getDFArgs(transformationDestination, transformationFileLocation, config.SourceMapping)
+	if err != nil {
+		k8s.logger.Errorw("Problem creating dataframe arguments", err)
+		return fmt.Errorf("error with getting df arguments %v", k8sArgs)
+	}
+	k8s.logger.Debugw("Running DF transformation", config)
+	if err := k8s.executor.ExecuteScript(k8sArgs); err != nil {
+		k8s.logger.Errorw("Error running dataframe job", err)
+		return fmt.Errorf("submit job for transformation %v failed to run: %v", config.TargetTableID, err)
+	}
+	k8s.logger.Debugw("Succesfully ran DF transformation", config)
 	return nil
 }
 
+func (k8s *K8sOfflineStore) updateQuery(query string, mapping []SourceMapping) (string, []string, error) {
+	sources := make([]string, len(mapping))
+	replacements := make([]string, len(mapping)*2) // It's times 2 because each replacement will be a pair; (original, replacedValue)
+
+	for i, m := range mapping {
+		replacements = append(replacements, m.Template)
+		replacements = append(replacements, fmt.Sprintf("source_%v", i))
+
+		sourcePath := ""
+		// sourcePath, err := k8s.getSourcePath(m.Source)
```

Comment:
Any reason why it's commented? Also, would all the sources just be empty string then?